Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp

```diff
@@ -104,16 +104,22 @@ void BitmapTexture::reset(const IntSize& size, OptionSet<Flags> flags)
     }
 }
 
-void BitmapTexture::updateContents(const void* srcData, const IntRect& targetRect, const IntPoint& sourceOffset, int bytesPerLine)
+void BitmapTexture::updateContents(const void* srcData, const IntRect& targetRect, const IntPoint& sourceOffset, int bytesPerLine, PixelFormat pixelFormat)
 {
-    // We are updating a texture with format RGBA with content from a buffer that has BGRA format. Instead of turning BGRA
-    // into RGBA and then uploading it, we upload it as is. This causes the texture format to be RGBA but the content to be BGRA,
-    // so we mark the texture to convert the colors when painting the texture.
+    RELEASE_ASSERT(pixelFormat == PixelFormat::RGBA8 || pixelFormat == PixelFormat::BGRA8);
+
+    if (pixelFormat == PixelFormat::RGBA8)
+        m_colorConvertFlags = { };
```

Comment:
For example, a single pixel update with RGBA format causes the whole texture to be interpreted as RGBA, which is wrong and will cause visual errors.